dcicutils/s3_utils.py

```diff
@@ -260,52 +271,73 @@ def health_page_get(self, health_page_key: str, *, default: Any = None, force: b
+            #            fact self.url needs to always be set to a non-empty value by the time we reach here, and
+            #            probably raise an error otherwise. We should also raise an error if there's a server
+            #            in the data we retrieved because it's just not going to get used.  -kmp 1-Mar-2023
             keys['server'] = self.url
-        return keys
+        return keys  # Might still have no server if self.url was not a URL
 
-    def get_ff_key(self):
+    def get_ff_key(self) -> ServerAuthDict:
+        # TODO: The return value should be just AuthDict if we decide get_access_keys() can't guarantee ServerAuthDict.
         return self.get_access_keys()
 
-    def get_higlass_key(self):
+    def get_higlass_key(self) -> Union[AnyJsonData, str]:  # parsed json or string that won't parse
+        # TODO: It would be nice if this could be returning something with a better promise. -kmp 1-Mar-2023
         # higlass key corresponds to Django server super user credentials
         return self.get_key(keyfile_name='api_key_higlass')
 
-    def get_google_key(self):
+    def get_google_key(self) -> Union[AnyJsonData, str]:  # parsed json or string that won't parse
+        # TODO: It would be nice if this could be returning something with a better promise. -kmp 1-Mar-2023
         return self.get_key(keyfile_name='api_key_google')
 
-    def get_jupyterhub_key(self):
+    def get_jupyterhub_key(self) -> Union[AnyJsonData, str]:  # parsed json or string that won't parse
+        # TODO: It would be nice if this could be returning something with a better promise. -kmp 1-Mar-2023
         # jupyterhub key is a Jupyterhub API token
         return self.get_key(keyfile_name='api_key_jupyterhub')
 
-    def get_key(self, keyfile_name='access_key_admin'):
+    def get_key(self, keyfile_name: str = 'access_key_admin'
+                ) -> Union[AnyJsonData, str]:  # parsed json or string that won't parse
         # Share secret encrypted S3 File
         response = self.s3.get_object(Bucket=self.sys_bucket,
                                       Key=keyfile_name,
                                       SSECustomerKey=os.environ['S3_ENCRYPT_KEY'],
                                       SSECustomerAlgorithm='AES256')
-        akey = response['Body'].read()
+        akey: Union[bytes, str] = response['Body'].read()
         if type(akey) == bytes:
-            akey = akey.decode()
+            akey: str = akey.decode()
         try:
             return json.loads(akey)
         except (ValueError, TypeError):
+            # TODO: Is there really a use case for this returning non-JSON data? If so, can we make a different
+            #       function that does this? Because this is just going to return garbage when peopla are expecting
+            #       key dictionaries and we really should raise an error in that case. -kmp 1-Mar-2023
```

Comment:
Yes there is, in the case of Higlass or JH keys.

Reply:
`access_key_foursight` follows similar patterns. No evidence that other than a dictionary is expected.